topcoat-data/topcoat-public
# Chore/add display value option to TTagsSelector

Goal: Give users the ability to include extra data in the filter dropdowns like CWE/CVE ID and title. After conversation with Ezra, the best way to achieve that in a flexible way is in the layer. So the TTagsSelector component will check for a DisplayValue column in the layer and show that instead of the key or value if it is present. Additionally, this approach will prevent having to update all of the TTagsSelectors where the additional functionality is not needed.

To test:

- checkout the meg-dev branch of topcoat-reports
- run a build and sync modules
- open Issues Detail page 
- add the CWE filter and  verify that the cwe IDs in the filter dropdown are all followed by " SQL Injection"
- add the CVE filter and  verify that the cve IDs in the filter dropdown are all followed by " SQL Injection"

General thoughts on this PR: It has been my experience that the less the data is manipulated in a component the fewer hacky things you have to do like manipulate date -> show it -> use that manipulated data to go find the original data to show something else. However attempting to refactor the TTagsSelector while preserving the url filters was not actually simplifying much and was therefor not worth the effort to continue. If y'all disagree I'm happy to continue with that approach instead.

[Pitch](https://www.notion.so/snyk/Enable-CVE-CWE-filter-to-include-the-description-for-easier-searching-c78f93373641466481c47fe6a471711f)
